cloud-info/Dockerfile

```diff
@@ -1,33 +1,57 @@
+FROM python:3 as build
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# hadolint ignore=DL3008
+RUN curl -s https://dist.eugridpma.info/distribution/igtf/current/GPG-KEY-EUGridPMA-RPM-3 \
+	| apt-key add - \
+    && echo "deb https://repository.egi.eu/sw/production/cas/1/current egi-igtf core" > /etc/apt/sources.list.d/igtf.list \
+    && apt-get update \
+    && apt-get install -y ca-policy-egi-core \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /cloud-info
+
+RUN python -m venv /cloud-info/venv
+ENV PATH="/cloud-info/venv/bin:$PATH"
+
+COPY requirements.txt .
+
+RUN pip install --no-cache-dir -r requirements.txt \
+    && cat /etc/grid-security/certificates/*.pem >> "$(python -m requests.certs)"
+
+COPY . .
+
+RUN pip install --no-cache-dir .
+
+# The actual image
 FROM python:3
 
 LABEL org.opencontainers.image.source=https://github.com/EGI-Federation/fedcloud-catchall-operations
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
-RUN mkdir /cloud-info
-COPY requirements.txt /cloud-info/requirements.txt
-RUN pip install --no-cache-dir -r /cloud-info/requirements.txt
-
-# CA certificates: install and add to python
 # hadolint ignore=DL3015, DL3008
-RUN curl -Ls \
-        https://dist.eugridpma.info/distribution/igtf/current/GPG-KEY-EUGridPMA-RPM-3 \
-        | apt-key add - \
-    && echo 'deb http://repository.egi.eu/sw/production/cas/1/current egi-igtf core' \
-        > /etc/apt/sources.list.d/cas.list \
-    && apt-get update \
-    && apt-get install -y jq \
-    && apt-get install -y ca-policy-egi-core \
-    && rm -rf /var/lib/apt/lists/* \
-    && cat /etc/grid-security/certificates/*.pem >> "$(python -m requests.certs)"
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+       jq rclone \
+    && rm -rf /var/lib/apt/lists/*
 
 
-COPY . /cloud-info/
-RUN pip install --no-cache-dir /cloud-info
+RUN mkdir /cloud-info \
+    && groupadd -g 1999 python \
+    && useradd -r -u 1999 -g python python \
+    && chown -R python:python /cloud-info
 
+WORKDIR /cloud-info
+
+COPY --chown=python:python --from=build /cloud-info/venv ./venv
 COPY ams-wrapper.sh /usr/local/bin/ams-wrapper.sh
```

Comment:
I would add a few comments to these ADD lines to explain what these files are, what they are _for_ and where to find documentation for them.